### DISCUSSION TITLE
iOS 에서 썸네일 저장시 이미지 반영 안됨

### DISCUSSION THREAD

**Discussion by da-in** [Yangjaecheon-Otter-Guardians/simple-thumbnail#108] [Bugs] [answered]
## Bugs Description

1. **버그 발생한 OS**  
   iphone 12 pro
2. **버그 상황 설명**
   iOS에서 랜덤이미지, 직접 업로드 이미지를 배경으로 사용할 경우, 다운로드 결과물에 이미지가 보이지 않습니다.

## BDD-Task!

- [ ] 랜덤 이미지 배경 적용 후 다운로드시 정상적으로 이미지가 보인다.
- [ ] 직접 업로드 배경 적용 후 다운로드시 정상적으로 이미지가 보인다.

**Comment by headring** [comment]
아이폰 가진 사람으로 확인해보니까 이미지 확장자 이름이 `.heic`도 있는 데 `.heif`도 있었어요.
그리고 갤럭시도 확인해보니까 고화질(?)(특히 야간사진)은 `heic`로 저장이 됩니다.

**Comment by da-in** [comment]
1. 데스크탑에서 색상배경 > 그라데이션 탭을 선택학 후
2. 직접 업로드로 들어가 이미지를 업로드 하면 안보임
* preview 의 배경 로직 문제

**Comment by da-in** [comment, marked as answer]
관련 이슈들입니다. 근본적인 해결법이라 우회적인 방식으로, 좋지 못합니다.  
모두가 `html-to-canvas` 를 사용하는 것을 근본적인 해결책이라 말합니다 ㅋㅋㅋㅋㅋ😆

일단 아래 구현으로 적용해두되, 추후 리팩토링에서 `html-to-canvas`를 사용하면 좋을 것 같습니다.

```js
toPng(previewRef.current).then(() => {
   if (previewRef.current) {
      toPng(previewRef.current).then(() => {
         if (previewRef.current) {
            toPng(previewRef.current).then((dataUrl) => {
               download(dataUrl, 'thumbnail.png');
            });
         }
      });
   }
});
```

**html-to-image**
https://github.com/bubkoo/html-to-image/issues/52

**dom-to-image**
https://github.com/tsayen/dom-to-image/issues/343
